### PR TITLE
Add test vectors for argon2id

### PR DESCRIPTION
**Pull request by lmctv**
Also documenting how both these and the argon2i vectors are generated (helps with#305)

### CODE REVIEW

**Comment by lmctv** [comment]
@alex , @reaperhulk these are just the test vectors generation docs and the generated vectors for the argon2id construction; as soon as they get in I can rebase my argon2id WIP branch, and hopefully getting it finished will be a matter of hours.

File: docs/vectors/argon2_vectors.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Code for the vector generator driver
+""""""""""""""""""""""""""""""""""""
+
+The code for ``argondriver.py`` is available inside
+the ``docs/vectors/python`` directory of PyNaCl distribution
+and can also be directly downloaded from
+:download:`argondriver.py <./python/argondriver.py>`.
+
+..
+    put the and... sentence under a ..only:: builder_html
```
Comment (reaperhulk) [inline docs/vectors/argon2_vectors.rst]: What's going on here?

File: docs/vectors/argon2_vectors.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+hex-encoded raw hash data.
+
+Using the default settings, the driver generates hex-encoded raw hash
+data; setting the ``-e`` option on the command line allows generation
+of modular crypt formatted data.
+
+The other command line options influence the minimum and maximum sizes
+of generated parameters as shown in the driver's command line help:
+
+.. code-block:: bash
```
Comment (reaperhulk) [inline docs/vectors/argon2_vectors.rst]: I don't think there's much value in showing the help output of the driver.

File: docs/vectors/argon2_vectors.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+
+Vector generation
+"""""""""""""""""
+
+The ``argondriver.py`` requires setting, via the command line option
+``-x``, the path to the argon2 executable, and the contruct to use
+in generating test data, which mush be either ``argon2i`` or ``argon2id``.
+
+and as a default generates
```
Comment (reaperhulk) [inline docs/vectors/argon2_vectors.rst]: What's going on here?

File: docs/vectors/python/argondriver.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+import argparse
+import json
+import random
+import string
+import subprocess
+import sys
+
+
+class argonRunner(object):
```
Comment (reaperhulk) [inline docs/vectors/python/argondriver.py]: This is an awfully involved class for something that really just invokes a subprocess in a loop. Do we really need something this complex? Is it taken from somewhere else?

File: docs/vectors/python/argondriver.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    p.add_argument('-M', '--max-memory-exponent', dest='mxmem',
+                   type=int, default=16)
+    p.add_argument('-t', '--min-time-opscount', dest='mniters',
+                   type=int, default=3)
+    p.add_argument('-T', '--max-time-opscount', dest='mxiters',
+                   type=int, default=3)
+    p.add_argument('-n', '--count', dest='n', type=int, default=10)
+    p.add_argument('-w', '--output', dest='outfile',
+                   default=sys.stdout, type=argparse.FileType('w'))
+    # p.add_argument('-r', '--input', dest='infile',
```
Comment (reaperhulk) [inline docs/vectors/python/argondriver.py]: We shouldn't be committing commented out code.

File: docs/vectors/argon2_vectors.rst (hunk 1 above)
Comment (lmctv) [reply docs/vectors/argon2_vectors.rst]: The download link does only make sense on html output, but if I write
```
The code for ``argondriver.py`` is available inside
￼the ``docs/vectors/python`` directory of PyNaCl distribution
..only:: builder_html
and can also be directly downloaded from
:download:`argondriver.py <./python/argondriver.py>`.
```
the download link line disappears from rtd's output. I left the commented-out note for letting others know the sentence output cannot be fixed until rtd starts enabling the ``builder_html`` tag while building in html format.

File: docs/vectors/python/argondriver.py (hunk 4 above)
Comment (lmctv) [reply docs/vectors/python/argondriver.py]: This is the same code I've put on a gist at the time of argon2i reference vectors generation, to which I just added the option to generate argon2id and argon2d password hashes. At least, adding the other hash constructions has been a no-brainer

**Pull request merged**